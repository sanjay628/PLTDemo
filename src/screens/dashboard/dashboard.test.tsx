import {fireEvent, render} from '@testing-library/react-native';
import axios from 'axios';
import React from 'react';
import {useSelector} from 'react-redux';
import DashboardScreen from './dashboard.screen';

const createTestProps = (props: Object) => ({
  ...props,
});
jest.mock('axios', () => ({
  get: jest.fn(),
}));
jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('DashboardScreen', () => {
  let props: any;
  beforeEach(() => {
    props = createTestProps({});
  });
  const mockShoppingList = [
    {id: 1, name: 'Item 1', qty: 2, price: 10},
    {id: 2, name: 'Item 2', qty: 1, price: 20},
  ];

  beforeEach(() => {
    (useSelector as jest.Mock).mockReturnValue({
      shoppingList: mockShoppingList,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render the DashboardScreen component correctly', async () => {
    const mockedAxios = axios as jest.Mocked<typeof axios>;
    mockedAxios.get.mockResolvedValueOnce({data: []});
    const {getByTestId, getByText} = render(<DashboardScreen />);
    expect(getByTestId('dashboard-screen')).toBeDefined();
    expect(getByText('Cart 2')).toBeDefined();
    await Promise.resolve();

    expect(mockedAxios.get).toHaveBeenCalledWith(
      'https://my-json-server.typicode.com/benirvingplt/products/products',
    );
  });

  it('should navigate to the cart screen when the cart view is pressed', () => {
    const navigationMock = {navigate: jest.fn()};
    const {getByTestId} = render(
      <DashboardScreen navigation={navigationMock} />,
    );
    const cartView = getByTestId('cart-view');
    fireEvent.press(cartView);
    expect(navigationMock.navigate).toHaveBeenCalledWith('CartScreen');
  });
  it('renders correctly', () => {
    const tree = render(<DashboardScreen {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
