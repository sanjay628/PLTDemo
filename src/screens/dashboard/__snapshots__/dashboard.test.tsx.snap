// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="dashboard-screen"
>
  <RCTScrollView
    data={Array []}
    extraData={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightblue",
        "borderRadius": 50,
        "bottom": "5%",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 40,
        "width": 160,
      }
    }
    testID="cart-view"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 32,
            "fontWeight": "600",
          },
          undefined,
          Object {
            "color": "#D32F2F",
          },
        ]
      }
    >
      Cart 2
    </Text>
  </View>
</RCTSafeAreaView>
`;
